jwt.js

```diff
@@ -39,7 +39,13 @@ function fastifyJwt (fastify, options, next) {
   let secretOrPrivateKey
   let secretOrPublicKey
 
-  if (typeof secret === 'object' && !Buffer.isBuffer(secret)) {
+  if (typeof secret === 'function' && secret.constructor.name === 'AsyncFunction') {
+    try {
+      secretOrPrivateKey = secretOrPublicKey = await secret()
+    } catch (err) {
+      return next(new Error('secret promise was rejected'))
+    }
```

Comment:
this code should be changed to reflect the earlier suggestion. don't use constructor.name to identify a function returning a promise because that works only if the function is async. you don't care whether the function is async, you care if the return value is a promise. you'll find examples of how to do this properly